src/validation/schema.js

```diff
@@ -154,6 +157,65 @@ const unwrapType = type => {
     : type
 }
 
+const gatherLocalTypes = defs =>
+  defs
+    .filter(
+      def =>
+        def.kind === 'ObjectTypeDefinition' ||
+        def.kind === 'EnumTypeDefinition' ||
+        def.kind === 'InterfaceTypeDefinition',
+    )
+    .map(def => def.name.value)
+
+const gatherForeignTypes = defs =>
+  defs
+    .filter(
+      def =>
+        def.kind === 'ObjectTypeDefinition' &&
+        def.name.value == RESERVED_TYPE &&
+        def.directives.find(
+          directive =>
+            directive.name.value == 'imports' &&
```

Comment:
I think the directive should be `@import`, not `@imports`. Similar to how in other languages you'd do `import ... from ...` and not `imports ... from ...`.